src/pages/AddPersonalBankAccountPage.tsx

```diff
@@ -16,69 +14,37 @@ import * as BankAccounts from '@userActions/BankAccounts';
-    plaidData: PlaidDataProps.plaidDataPropTypes,
+    plaidData: OnyxEntry<PlaidData>;
 
     /** The details about the Personal bank account we are adding saved in Onyx */
-    personalBankAccount: PropTypes.shape({
-        /** An error message to display to the user */
-        error: PropTypes.string,
-
-        /** Whether we should show the view that the bank account was successfully added */
-        shouldShowSuccess: PropTypes.bool,
-
-        /** Any reportID we should redirect to at the end of the flow */
-        exitReportID: PropTypes.string,
-
-        /** Whether we should continue with KYC at the end of the flow  */
-        shouldContinueKYCOnSuccess: PropTypes.bool,
-
-        /** Whether the form is loading */
-        isLoading: PropTypes.bool,
-
-        /** The account ID of the selected bank account from Plaid */
-        plaidAccountID: PropTypes.string,
-    }),
+    personalBankAccount: OnyxEntry<PersonalBankAccount>;
 };
 
-const defaultProps = {
-    plaidData: PlaidDataProps.plaidDataDefaultProps,
-    personalBankAccount: {
-        error: '',
-        shouldShowSuccess: false,
-        isLoading: false,
-        plaidAccountID: '',
-        exitReportID: '',
-        shouldContinueKYCOnSuccess: false,
-    },
-};
-
-function AddPersonalBankAccountPage({personalBankAccount, plaidData}) {
+function AddPersonalBankAccountPage({personalBankAccount, plaidData}: AddPersonalBankAccountPageWithOnyxProps) {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
     const [selectedPlaidAccountId, setSelectedPlaidAccountId] = useState('');
-    const shouldShowSuccess = lodashGet(personalBankAccount, 'shouldShowSuccess', false);
+    const shouldShowSuccess = personalBankAccount?.shouldShowSuccess ?? false;
 
-    /**
-     * @returns {Object}
-     */
     const validateBankAccountForm = () => ({});
 
     const submitBankAccountForm = useCallback(() => {
-        const selectedPlaidBankAccount = _.findWhere(lodashGet(plaidData, 'bankAccounts', []), {
-            plaidAccountID: selectedPlaidAccountId,
-        });
-
-        BankAccounts.addPersonalBankAccount(selectedPlaidBankAccount);
+        const bankAccounts = plaidData?.bankAccounts ?? [];
+        const selectedPlaidBankAccount = bankAccounts.find((bankAccount) => bankAccount.plaidAccountID === selectedPlaidAccountId);
+        if (selectedPlaidBankAccount) {
+            BankAccounts.addPersonalBankAccount(selectedPlaidBankAccount);
+        }
     }, [plaidData, selectedPlaidAccountId]);
 
     const exitFlow = useCallback(
         (shouldContinue = false) => {
-            const exitReportID = lodashGet(personalBankAccount, 'exitReportID');
-            const onSuccessFallbackRoute = lodashGet(personalBankAccount, 'onSuccessFallbackRoute', '');
+            const exitReportID = personalBankAccount?.exitReportID;
+            const onSuccessFallbackRoute: Route = (personalBankAccount?.onSuccessFallbackRoute as Route) ?? '';
```

Comment:
Could you try to update `onSuccessFallbackRoute` type to `Route` instead of assertion